### PR TITLE
Add ability to make an alpine image

### PR DESCRIPTION
**Pull request by flynnguy**
So this adds a `Dockerfile.alpine` and also adds some `Makefile` targets to build various images. I left the script that also builds it, I figure we should also talk about which approach we like better. Personally I like the Makefile targets better because you can chain things together and it's a common place to look for where to do what. We could also just have the `Makefile` targets use the script as well? But in this case, it seems simple enough to just use make.

### CODE REVIEW

**Comment by sarahbx** [comment]
If the scripts have any command line arguments, then make targets would be the best option. Simplifies the use to: `make thing`. CI commands are all `make thing` without having to modify usage in two places. Any time I've created a script, eventually it would get arguments added to it, regardless of my original intention.

**Review by sstelfox — COMMENTED** [review]
I'm not a huge fan of slashes in the make names, but a Makefile definitely makes sense (:D)

**Comment by flynnguy** [comment]
This script just builds and pushes the docker image. Since this is for a base image, I can't imagine it doing a whole lot more.

**Comment by flynnguy** [comment]
@sstelfox it's a convention I've used in the past to help organize things, we can drop it... especially here as there's not a whole lot going on currently.

**Comment by sstelfox** [comment]
> @sstelfox it's a convention I've used in the past to help organize things, we can drop it... especially here as there's not a whole lot going on currently.

It's entirely a personal preference to me and an opinion I don't hold strongly hahah. We don't have to change it, I'm just not personally a huge fan.

**Comment by tyler-sommer** [comment]
Makefile is better than the minefield that is writing bash scripts, right? +1 for Makefile

File: Dockerfile.alpine [hunk 1]
```diff
@@ -0,0 +1,19 @@
+FROM ruby:2.6.3-alpine3.9
+
+RUN apk --no-cache add curl nodejs nodejs-npm postgresql-client postgresql-dev fftw fftw-dev
```
Comment (sarahbx) [inline Dockerfile.alpine]: I think we may want to add `apk --no-cache upgrade && ..` at the beginning, make sure we get any security updates not in the base image

File: Makefile [hunk 2]
```diff
@@ -0,0 +1,34 @@
+VERSION?=$(shell git describe --tags 2> /dev/null || echo '0.0.0')
+DOCKER_IMAGE=minimsecure/ruby-docker-image
+
+.PHONY: image
+image:
+	docker build \
+		-t minimsecure/ruby-docker-image:$(VERSION) \
+		-t $(DOCKER_IMAGE):latest \
```
Comment (sarahbx) [inline Makefile:8]: Are we currently using `latest` when pulling images? Just wondering if this is a pattern we want to use/keep using. It's convenient, but can easily propagate a breaking change.

File: Dockerfile.alpine (hunk 1 above)
Comment (flynnguy) [reply Dockerfile.alpine]: 👍

File: Makefile (hunk 2 above)
Comment (flynnguy) [reply Makefile:8]: We are not currently using it to build retroelk:
https://github.com/violetatrium/retroelk/blob/master/.circleci/config.yml#L3

I think having `latest` is useful when developing but I agree that builds that get pushed out to any environment should not use it.

**Review by sarahbx — APPROVED** [review]

**Pull request merged**